server/src/test/java/io/crate/integrationtests/PgCatalogITest.java

```diff
@@ -328,4 +328,15 @@ public void test_pg_constrains_conkey_array_populated() throws Exception {
         ));
 
     }
+
+    @UseRandomizedSchema(random = false)
+    @Test
+    public void test_pg_class_oid_equals_cast_of_string_to_regclass() {
+        execute("CREATE TABLE persons (x INT)");
+        execute("SELECT " +
```

Comment:
This is not relevant to this PR:
Postgres behaves differently:
```
postgres=# SELECT '"marios"'::regclass oid_from_relname, oid FROM pg_class WHERE relname ='marios';
 oid_from_relname |  oid  
------------------+-------
 marios           | 24616
(1 row)

postgres=# SELECT '"marios"'::regclass::int oid_from_relname, oid FROM pg_class WHERE relname ='marios';
 oid_from_relname |  oid  
------------------+-------
            24616 | 24616
(1 row)
```

So, it doesn't do an implicit cast to int, should we follow this behaviour as well?

Reply:
At the moment we use the `regclass` type for oid's inside `pg_catalog` tables because we lack a dedicated `OID` type. That's why we default to return the `oid` of a regclass type at HTTP instead of the relation name like PG does.
Such we could only fix that once we implement a dedicated `OID` data type.